pkg/controller/persistentpodstate/persistent_pod_state_event_handler.go

```diff
@@ -41,9 +47,58 @@ var _ handler.EventHandler = &enqueueRequestForPod{}
 
 type enqueueRequestForPod struct {
 	reader client.Reader
+	client client.Client
 }
 
 func (p *enqueueRequestForPod) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	pod := evt.Object.(*corev1.Pod)
+	// 1. find pod's owners groupVersionKind
+	ownerReferencesGVK := make(map[schema.GroupVersionKind]struct{})
+	for _, owner := range pod.OwnerReferences {
+		gv, err := schema.ParseGroupVersion(owner.APIVersion)
+		if err != nil {
+			return
+		}
+
+		if owner.Kind == KruiseKindSts.Kind || owner.Kind == KindSts.Kind {
+			continue
+		}
+
+		ownerReferencesGVK[schema.GroupVersionKind{
+			Group:   gv.Group,
+			Version: gv.Version,
+			Kind:    owner.Kind,
+		}] = struct{}{}
+	}
+
+	whiteList, err := configuration.GetPPSWatchWatchCustomWorkloadWhiteList(p.client)
+	if err != nil {
+		klog.Errorf("Failed to get persistent pod state config white list, error: %v\n", err.Error())
+		return
+	}
+	if whiteList == nil {
+		return
+	}
+
+	// 2.add watch
+	for gvk := range ownerReferencesGVK {
+		enable := false
+		for _, sgv := range whiteList.Workloads {
+			if gvk.GroupKind().String() == sgv.String() {
+				enable = true
+				break
+			}
+		}
+		if !enable {
+			continue
+		}
+
+		_, err := util.AddWatcherDynamically(runtimeController, workloadHandler, gvk)
```

Comment:
It is recommended not to implement the watch dynamic crd method in the event handler, but in reconcile, please refer to: https://github.com/openkruise/rollouts/blob/master/pkg/controller/batchrelease/ batchrelease_controller.go#L165